src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceVariableService_IntroduceLocal.cs

```diff
@@ -173,70 +176,60 @@ private bool CanUseVar(ITypeSymbol typeSymbol)
             LocalDeclarationStatementSyntax declarationStatement,
             bool allOccurrences,
+            bool createReturnStatement,
             CancellationToken cancellationToken)
         {
             var oldBody = arrowExpression;
             var oldParentingNode = oldBody.Parent;
             var leadingTrivia = oldBody.GetLeadingTrivia()
                                        .AddRange(oldBody.ArrowToken.TrailingTrivia);
 
-            var newStatement = Rewrite(document, expression, newLocalName, document, oldBody.Expression, allOccurrences, cancellationToken);
-            var newBody = SyntaxFactory.Block(declarationStatement, SyntaxFactory.ReturnStatement(newStatement))
+            var newExpression = Rewrite(document, expression, newLocalName, document, oldBody.Expression, allOccurrences, cancellationToken);
+
+            var convertedStatement = createReturnStatement
+                ? SyntaxFactory.ReturnStatement(newExpression)
+                : SyntaxFactory.ExpressionStatement(newExpression) as StatementSyntax;
+
+            var newBody = SyntaxFactory.Block(declarationStatement, convertedStatement)
                                        .WithLeadingTrivia(leadingTrivia)
                                        .WithTrailingTrivia(oldBody.GetTrailingTrivia())
                                        .WithAdditionalAnnotations(Formatter.Annotation);
 
-            SyntaxNode newParentingNode = null;
-            if (oldParentingNode is BasePropertyDeclarationSyntax baseProperty)
+            SyntaxNode newParentingNode = oldParentingNode; // in case we fail for some reason, at least don't remove it!
+            
+            switch (oldParentingNode)
             {
-                var getAccessor = SyntaxFactory.AccessorDeclaration(SyntaxKind.GetAccessorDeclaration, newBody);
-                var accessorList = SyntaxFactory.AccessorList(SyntaxFactory.List(new[] { getAccessor }));
+                case BasePropertyDeclarationSyntax baseProperty:
+                    var getAccessor = SyntaxFactory.AccessorDeclaration(SyntaxKind.GetAccessorDeclaration, newBody);
+                    var accessorList = SyntaxFactory.AccessorList(SyntaxFactory.List(new[] { getAccessor }));
 
-                newParentingNode = baseProperty.RemoveNode(oldBody, SyntaxRemoveOptions.KeepNoTrivia);
-
-                if (newParentingNode.IsKind(SyntaxKind.PropertyDeclaration))
-                {
-                    var propertyDeclaration = ((PropertyDeclarationSyntax)newParentingNode);
-                    newParentingNode = propertyDeclaration
+                    newParentingNode = baseProperty
+                        .RemoveNode(oldBody, SyntaxRemoveOptions.KeepNoTrivia)
                         .WithAccessorList(accessorList)
+                        .TryWithSemicolonToken(SyntaxFactory.Token(SyntaxKind.None))
+                        .WithTrailingTrivia(baseProperty.TryGetSemicolonToken().TrailingTrivia);
```

Comment:
this feels strange.  why not just do:   baseProperty.WithoutTrivia().WithAccessorList(...).TryWith()...WithTriviaFrom(baseProperty)?

Reply:
i don't know, this is the original code, I tried to only change what I needed and not question it

Reply:
IN this case, since you're changing it, i think it would be beneficial to just do things in a cleaner way if possible.

Reply:
OK

Reply:
Discussed in gitter.  What i suggested would lead to a change in behavior.  That change would likely be *beneficial*.  But to keep the scope of this PR down, it doesn't have to happen here.

Reply:
it wouldn't, I was talking about making use of [ArrowExpressionClauseSyntaxExtensions.TryConvertToBlock](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/Extensions/ArrowExpressionClauseSyntaxExtensions.cs,bb53b4cb5bf05658). That would bring consistency with other code fixes that expend expression bodies into blocks, but would change behavior. Also it's called "Try" because in some cases with preprocessor directives it will just return false - this can't be expanded. I don't know what that should even do here. Not offer "inline variable"?

Reply:
>  I don't know what that should even do here. Not offer "inline variable"?

Seems acceptable.  If you can't make the transformation, then you can't really offer the feature.

> I would definitely like to get consistent behavior across these fixes but I don't know if I can do it here.

That seems fine.

> And there are questions to be answered about which behavior to pick. And what to do about the directives.

Do you have any specific questions?  If so, we can attempt to answer them here (or gitter, or through some other issue).